rpc/rpcserver/server.go

```diff
@@ -796,18 +796,64 @@ func (s *walletServer) SweepAccount(ctx context.Context, req *pb.SweepAccountReq
 		return nil, translateError(err)
 	}
 
-	var txBuf bytes.Buffer
-	txBuf.Grow(tx.Tx.SerializeSize())
-	err = tx.Tx.Serialize(&txBuf)
-	if err != nil {
-		return nil, translateError(err)
+	sweepTxs := make([]*txauthor.AuthoredTx, 0)
+	maxTxSize := s.wallet.ChainParams().MaxTxSize
+	if tx.EstimatedSignedSerializeSize > maxTxSize {
+		var prevTx *txauthor.AuthoredTx
+		outs := make([]*wire.TxOut, 0)
+		sweepOuts := tx.Tx.TxOut
+
+		for idx := 0; idx < len(sweepOuts); idx++ {
+			outs = append(outs, sweepOuts[idx])
+			currTx, err := s.wallet.NewUnsignedTransaction(outs, feePerKb,
+				account, int32(req.RequiredConfirmations),
```

Comment:
This creates double spends due to using the same account as the funding source in a loop with no knowledge of previous picked inputs.  When moving this to the wallet module, a funding source wrapper should be used to prevent duplicate inputs being picked.